source/templates/layout.html

```diff
@@ -237,6 +237,27 @@
             title="{% trans docstitle=docstitle|e %}Search within {{ docstitle }}{% endtrans %}"
             href="{{ pathto('_static/opensearch.xml', 1) }}"/>
       {%- endif %}
+
+      {# Fragment redirects #}
+      {%- if has_fragment_redirects %}
+      <script>
+        {{ fragment_redirects }}
+        const url = new URL(window.location.href);
+        if (url.hash !== "" && url.hash !== "#") {
+          const fragment = url.hash.substring(1);
+          if (fragment in redirects) {
```

Comment:
Just to confirm: `redirects` is the path redirects we've defined in the config, right?

Reply:
I think it'd be valuable to add some input validation here since we're taking that fragment/substring directly from the end-user.

Reply:
I've added fragment validation into both `layout.html` and `redirect.html`